### ISSUE TITLE
"object" not callable from functools.partial(print, file=...)

### ISSUE THREAD

**Issue by mgedmin** [python/mypy#8928]
This feels like a bug to me:

```python
import functools
from typing import Optional, TextIO


def report(file: Optional[TextIO]) -> None:
    print_ = functools.partial(print, file=file) if file is not None else print
    print_("hello")
```

mypy reports

```
mypy_partial_print.py:7: error: "object" not callable
```

I would expect it to accept the code.

mypy version is 0.770.  git master (commit 273a86557e6e76acd52e6588230aa8e4ac5de532) reports the same error.

Note that mypy accepts slight variations, e.g.

```python
def report(file: TextIO) -> None:
    print_ = functools.partial(print, file=file)
    print_("hello")
```

is fine.

`reveal_type(print_)` in the original example says it's a `builtins.object`.  It looks like mypy cannot reconcile a function with its own partial application.

**Comment by mgedmin** [comment]
Hm, this looked different enough from #1484 that I decided to report it as a separate issue, but now I've found #2779 which is similar enough to be a duplicate, and I see that it was closed in favor of #1484.

**Comment by JukkaL** [comment]
This is basically the same issue as #2779. Closing as duplicate.
